src/data/nat/digits.lean

```diff
@@ -444,3 +444,28 @@ end
 
 lemma le_digits_len_le (b n m : ℕ) (h : n ≤ m) : (digits b n).length ≤ (digits b m).length :=
 monotone_of_monotone_nat (digits_len_le_digits_len_succ b) h
+
+lemma digits_ge_base_pow_len (b m : ℕ) : m ≠ 0 → (b + 2) ^ ((digits (b + 2) m).length) ≤ (b + 2) * m :=
```

Comment:
Can you replace `b+2` with `b` and add the hypothesis `2 <= b`. That is often a more useful form.

Reply:
I think this version can be kept as-is, but as a primed version. So `digits_ge_base_pow_len'`. Then the version Floris is suggesting can become the unprimed version.

Angela, see `digits_lt_base` or `of_digits_lt_base_pow_length` or one of the other lemmas if you're unsure how to do this.

Reply:
Okay thank you so much Shing!